testing/sdk_auth.py

```diff
@@ -142,35 +97,31 @@ def _get_master_public_ip() -> str:
     """
     dcos_url, headers = sdk_security.get_dcos_credentials()
     cluster_metadata_url = "{cluster_url}/metadata".format(cluster_url=dcos_url)
-    raw_response = sdk_cmd.request("GET", cluster_metadata_url, verify=False)
-    if not raw_response.ok:
-        raise RuntimeError("Unable to get the master node's public IP address: {err}".format(err=repr(raw_response)))
+    response = sdk_cmd.request("GET", cluster_metadata_url, verify=False)
+    if not response.ok:
+        raise RuntimeError("Unable to get the master node's public IP address: {err}".format(err=repr(response)))
 
-    response = json.loads(raw_response.text)
+    response = response.json()
     if "PUBLIC_IPV4" not in response:
         raise KeyError("Cluster metadata does not include master's public ip: {response}".format(
-            response=repr(raw_response)))
+            response=repr(response)))
 
     public_ip = response["PUBLIC_IPV4"]
     log.info("Master public ip is {public_ip}".format(public_ip=public_ip))
     return public_ip
 
 
-def _create_temp_working_dir() -> str:
+def _create_temp_working_dir() -> TemporaryDirectory:
     """
     Creates a temporary working directory to enable setup of the Kerberos environment.
-    :return (str): The path of the working directory.
+    :return (TemporaryDirectory): The TemporaryDirectory object holding the context of the temp dir.
     """
-    dir_name = "/tmp/{kerberos}_{current_time}".format(kerberos=KERBEROS_APP_ID, current_time=str(time.time()))
-    log.info("Creating temp working directory {}".format(dir_name))
-    try:
-        os.makedirs(dir_name)
-    except OSError as e:
-        raise OSError("Can't create temp working directory: {}".format(repr(e)))
-
-    return dir_name
+    tmp_dir = TemporaryDirectory()
+    log.info("Created temp working directory {}".format(tmp_dir.name))
+    return tmp_dir
 
 
+#TODO: make this generic and put in sdk_utils.py
```

Comment:
I did a little spelunking in the API docs for the Mesos file api as well as the DC/OS CLI code that uses it to do task logging.

I think we can actually pretty easily swap this to the file api:
<cluster-url>/slave/<agent-id>/files/read.json?path=<url-encoded-absolute-path>

I tested this against your cluster and was able to download the `keytab` file from the KDC with no problems.

Reply:
Changed to use the files API.